src/sdk/pynni/nni/nas/pytorch/fixed.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+import torch
+
+from nni.nas.pytorch.mutables import MutableScope
+from nni.nas.pytorch.mutator import Mutator
+
+
+class FixedArchitecture(Mutator):
+
+    def __init__(self, model, fixed_arc, strict=True):
+        """
+        Initialize a fixed architecture mutator.
+
+        Parameters
+        ----------
+        model: nn.Module
+            A mutable network.
+        fixed_arc: str or dict
+            Path to the architecture checkpoint (a string), or preloaded architecture object (a dict).
+        strict: bool
+            Force everything that appears in `fixed_arc` to be used at least once.
+        """
+        super().__init__(model)
+        self._fixed_arc = fixed_arc
+
+        mutable_keys = set([mutable.key for mutable in self.mutables if not isinstance(mutable, MutableScope)])
+        fixed_arc_keys = set(self._fixed_arc.keys())
+        if fixed_arc_keys - mutable_keys:
```

Comment:
can you explain the line 29?

Reply:
It's to check the matching between parsed search space and parameters stored in file and throws friendly error messages.